pkg/envoy/server.go

```diff
@@ -702,22 +855,35 @@ func getDirectionNetworkPolicy(l4Policy policy.L4PolicyMap, policyEnforced bool)
 		}
 
 		allowAll := false
+		// Assume none of the rules have side-effects so that rule evaluation can
+		// be stopped as soon as the first allowing rule is found. 'canShortCircuit'
+		// is set to 'false' below if any rules with side effects are encountered,
+		// causing all the applicable rules to be evaluated instead.
+		canShortCircuit := true
 		for sel, l7 := range l4.L7RulesPerEp {
-			rule := getPortNetworkPolicyRule(sel, l4.L7Parser, l7)
+			rule, cs := getPortNetworkPolicyRule(sel, l4.L7Parser, l7)
+			log.Debugf("Policy %s: %s can short circuit: %v (canShortCircuit %v)", sel.String(), rule.String(), cs, canShortCircuit)
+			if !cs {
+				canShortCircuit = false
+			}
 			if rule != nil {
 				if len(rule.RemotePolicies) == 0 && rule.L7 == nil {
 					// Got an allow-all rule, which would short-circuit all of
 					// the other rules. Just set no rules, which has the same
 					// effect of allowing all.
 					allowAll = true
-					pnp.Rules = nil
-					break
 				}
 
 				pnp.Rules = append(pnp.Rules, rule)
 			}
 		}
 
+		// Short-circuit rules if a rule allows all and all other rules can be short-circuited
+		if allowAll && canShortCircuit {
+			log.Info("Short circuiting HTTP rules due to rule allowing all and no other rules needing attention")
```

Comment:
This looks like a debug-level log, I don't think it's meaningful to most users and it's prone to repeating itself in the logs when a policy is evaluated.

Reply:
Right, changed to `Debug`